scripts/SREP/etcd-complete-health-check/script.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+echo "[INFO] Complete ETCD Status"
+
+## Get the ETCD pod
+ETCD_POD=$(oc get pods -l k8s-app=etcd -n openshift-etcd -o jsonpath='{.items[*].metadata.name}' | awk '{ print $1 }')
+
+## Print out the member table
+echo
+echo "[INFO] etcdctl member list"
+oc exec "$ETCD_POD" -n openshift-etcd -c etcdctl -- sh -c "etcdctl member list -w table"
+
+## Print out the summary of the etcd members
+echo
+echo "[INFO] summary of etcd"
+ETCD_SUMMARY=$(oc get etcd -o=jsonpath='{range .items[0].status.conditions[?(@.type=="EtcdMembersAvailable")]}{.message}{"\n"}')
+echo "$ETCD_SUMMARY"
+
+## Print out the summary of Endpoint Status
+echo
+echo "[INFO] etcdctl endpoint status"
+endpoint_status=$(oc exec "$ETCD_POD" -n openshift-etcd -c etcdctl -- sh -c "etcdctl endpoint status -w table" 2>/dev/null)
+echo "$endpoint_status" 
+
+
+## Print out the summary of Endpoint Health
+echo
+echo "[INFO] etcdctl endpoint health"
+endpoint_health=$(oc exec "$ETCD_POD" -n openshift-etcd -c etcdctl -- sh -c "etcdctl endpoint health -w table" 2>/dev/null)
+echo "$endpoint_health"
+
+
+# Get the logs for the unhealthy node(s)
+echo "$ETCD_SUMMARY" | sed -n 1'p' | tr ',' '\n' | grep unhealthy  |  while read word; do
+    name=$(echo "$word" | awk '{ print $1 }')
+    echo
+    echo "[INFO] logs for $name"
+    echo   
+    if command=$(oc logs -c etcd -n openshift-etcd etcd-"$name")
+    then
+        echo "$command"
+    elif cmd=$(oc get nodes | grep master | awk '{print $1}' | sed 1q |
+        while IFS= read -r nodename; 
+            do 
+                echo $'\nUsing crictl to fetch logs\n'
+                oc -n default debug node/"$nodename" -- sh -c 'chroot /host crictl ps -aql --label "io.kubernetes.container.name=etcd" | chroot /host xargs crictl logs --since 48h'; 
```

Comment:
This could end up being quite a lot of logs and threatens to overwhelm all the other information the script returns. I would recommend putting an upper bound on how many it returns with a `--tail <lines>` parameter to the `crictl logs` and limiting that to a small-ish number.